Client/Frontend/Login Management/LoginListViewController.swift

```diff
@@ -207,7 +208,10 @@ private class LoginCursorDataSource: NSObject, UITableViewDataSource {
         }
 
         let titleForSectionAtIndex = sectionTitles[section]
-        let logins = cursor?.filter { $0?.hostname.asURL?.host?.uppercaseString.startsWith(titleForSectionAtIndex) ?? false }
-        return logins?.flatMap { $0 } ?? []
+        let logins = cursor?.filter { $0?.hostname.asURL?.baseDomain()?.uppercaseString.startsWith(titleForSectionAtIndex) ?? false }
+        let flattenLogins = logins?.flatMap { $0 } ?? []
+        return flattenLogins.sort { login1, login2 in
+            login1.hostname.asURL?.baseDomain() < login2.hostname.asURL?.baseDomain()
```

Comment:
Yup that makes more sense. So the ordering would be:

a.b.f.domain.com
a.b.c.domain.com

First check baseDomain (domain). Then check sub domains left -> right (a.b.f vs a.b.c) resulting in

a.b.c.domain.com
a.b.f.domain.com